tests/h/session_test.py

```diff
@@ -85,14 +141,106 @@ def test_profile_sorts_groups():
         FakeGroup('b', 'Group B'),
         FakeGroup('a', 'Group B'),
     ]
-    request = mock.Mock(authenticated_user=fake_user)
+    request = mock.Mock(authenticated_user=fake_user,
+                        auth_domain=fake_user.authority)
     profile = session.profile(request)
 
     ids = [group['id'] for group in profile['groups']]
 
     assert ids == ['__world__', 'c', 'a', 'b']
 
 
+def test_world_group_in_authenticated_profile(fake_user, auth_domain):
+    request = mock.Mock(authenticated_user=fake_user, auth_domain=auth_domain)
+
+    result = session.profile(request)
+
+    assert '__world__' in [g['id'] for g in result['groups']]
+
+
+def test_world_group_in_anonymous_profile(auth_domain):
+    request = mock.Mock(authenticated_user=None, auth_domain=auth_domain)
+
+    result = session.profile(request)
+
+    assert '__world__' in [g['id'] for g in result['groups']]
+
+
+def test_world_group_not_in_third_party_profile(third_party_user, auth_domain):
+    request = mock.Mock(authenticated_user=third_party_user,
+                        auth_domain=auth_domain)
+
+    result = session.profile(request)
+
+    assert '__world__' not in [g['id'] for g in result['groups']]
+
+
+def test_world_group_is_public_in_profile(fake_user, auth_domain):
+    request = mock.Mock(authenticated_user=fake_user, auth_domain=auth_domain)
+
+    profile = session.profile(request)
+    world_group = [g for g in profile['groups'] if g['id'] == '__world__'][0]
+
+    assert world_group['public'] is True
+
+
+def test_private_group_is_not_public_in_profile(auth_domain):
```

Comment:
Some of these tests for `session.profile` look like they are repeating the tests for `session.model` quite a bit. Given that `h.session` is exactly the same as `h.profile` except for the addition of the `csrf` token, perhaps you could simplify the `session` tests to just checking that they return the value of `profile` with that one parameter added?